server/controllers/user.controller.ts

```diff
@@ -1,30 +1,134 @@
-const {JWT} = require('../token/jwt');
+import { CommandCompleteMessage } from "pg-protocol/dist/messages";
+
+const JWT = require('../token/jwt');
 const bcrypt = require('bcrypt');
 const database = require('../db/index');
 const router = require('express').Router();
 
 class UserController {
 
-     async login(req, res){
-        let user = await this.getUserByEmail(req.body.email);
-        if (!user){
+     async registerUser(email:string, password:string, isAdmin:boolean){
+        // return True if register is possible (all conditions are met)
+        // otherwise, return False
+
+        // Standard email format 
+        if (! email.match(/^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/))
+            return {ok: false, message: "Invalid email format!"};
+
+        // Minimum eight characters, at least one uppercase letter, one lowercase letter, one number and one special character    
+        if (! password.match(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/))
+            return {ok: false, message: "Invalid password format!"};
+        
+        // The isAdmin field shoul be checked 
+        if (isAdmin  == undefined)  
+            return {ok: false, message: "Undefined isAdmin!"}; 
+
+        return {ok: true, message: "Valid fields!"};;      
+     } 
+
+     async register(req, res){
+
+        // check if user doesn't exist already
+        const user = await this.getUserByEmail(req.body.email);
+        if (user) {
             return res.status(400).json({
-                message: 'Invalid email!',
+                message: 'User already exists!',
             });
         }
 
-        const validUser = await this.verifyUser(req.body.password, user.password);
-        if (!validUser){
-            return res.status(400).json({
-                message: 'Invalid password!',
+        const validUser =  await this.registerUser(req.body.email, req.body.password, req.body.is_admin);
+        console.log(validUser.message);
+        if (! validUser.ok){
+            // verify the completed fields
+            return res.status(422).json({
+                message: 'Invalid input!'
             });
         }
 
-        const token = JWT.createAccessToken({ id: user.id }); //FIXME:  TypeError: Cannot read property 'createAccessToken' of undefined
+        else{
```

Comment:
It isn't important very now(so it's perfectly fine to leave it like that), but I think it's useful to know: the _else_ branch is not necessary here because in the previous _if_ block you've used a _return_ statement at the end. That is called an _early return_.

So,

```ts
if (condition) {
 ...
 return 'foo';
} else {
 console.log('else branch')
}
```

can be replaced with:

```ts
if (condition) {
 ...
 return 'foo';
} 

console.log('else branch')
```